overlord/snapstate/snapstate_test.go

```diff
@@ -5760,13 +5783,14 @@ func (s *snapmgrTestSuite) TestRemoveMany(c *C) {
 	}
 }
 
-func taskWithKind(ts *state.TaskSet, kind string) *state.Task {
+func findTasks(ts *state.TaskSet, pred func(task *state.Task) bool) []*state.Task {
+	var tasks []*state.Task
 	for _, task := range ts.Tasks() {
-		if task.Kind() == kind {
-			return task
+		if pred(task) {
```

Comment:
Why moving the predicate out when there's only a single predicate we care about?  Feels like early-generalization. The test looks simpler and more readable when we name this as tasksWithKind and don't ask for a predicate.